.circleci/continue_config.yml

```diff
@@ -0,0 +1,199 @@
+version: 2.1
+
+orbs:
+  codecov: codecov/codecov@3.2.5
+
+parameters:
+  engine-changed:
+    type: boolean
+    default: false
+  webrtc-changed:
+    type: boolean
+    default: false
+  hls-changed:
+    type: boolean
+    default: false
+  rtsp-changed:
+    type: boolean
+    default: false
+  integration-changed:
+    type: boolean
+    default: false
+
+jobs:
+  test_package:
+    parameters:
+      workdir:
+        type: string
+      codecov_flag:
+        type: string
+      mix_test_args:
+        type: string
+        default: ""
+      upgrade_packages:
+        type: boolean
+        default: false
+
+    resource_class: large
+    docker:
+      - image: membraneframeworklabs/docker_membrane:v1.8.1
+        environment:
+          MIX_ENV: test
+
+    working_directory: << parameters.workdir >>
+
+    steps:
+      - checkout:
+          path: ~/app
+      - when:
+          condition: << parameters.upgrade_packages >>
+          steps:
+            - run: apt update -y && apt upgrade -y
+      - run: mix deps.get
+      - run: mix coveralls.json << parameters.mix_test_args >>
+      - codecov/upload:
+          flags: << parameters.codecov_flag >>
+
+  lint_package:
+    parameters:
+      workdir:
+        type: string
+      disable_dev_checks:
+        type: boolean
+        default: false
+
+    resource_class: large
+    docker:
+      - image: membraneframeworklabs/docker_membrane:v1.8.1
+        environment:
+          MIX_ENV: dev
+
+    working_directory: << parameters.workdir >>
+
+    steps:
+      - checkout:
+          path: ~/app
+      - run: mix deps.get
+      - run: mix format --check-formatted
+      - unless:
+          condition: << parameters.disable_dev_checks >>
+          steps:
+            - run: mix compile --force --warnings-as-errors
+            - run: mix credo
+            - run: mix dialyzer
+            - run: mix docs && mix docs 2>&1 | (! grep -q "warning:")
+
+  test_webrtc_integration:
+    resource_class: large
+    docker:
+      - image: membraneframeworklabs/docker_membrane:v1.8.1
+        environment:
+          MIX_ENV: test
+          NODE_ENV: development
+        user: root
+
+    working_directory: ~/app/membrane_rtc_engine
+
+    steps:
+      - checkout:
+          path: ~/app
+      # TODO remove this after updating Node in out docker_membrane repository
+      - run: npm install npm@9.2.0 -g
+      - run: mix test.webrtc.integration
+
+
+workflows:
+  test-and-lint-engine:
```

Comment:
Shouldn't this be done in two separate steps?

Reply:
It's a workflow, it's still gonna run two separate jobs (test_engine and lint_engine). As these jobs are always run together, I chose to aggregate them under one workflow